File: jambeth/jAmbeth.Bytecode/src/de/osthus/ambeth/bytecode/IBytecodeClassLoader.java
package de.osthus.ambeth.bytecode;

import java.io.Writer;

public interface IBytecodeClassLoader extends IBytecodePrinter
{
	byte[] readTypeAsBinary(Class<?> type);

	String getBytecodeTypeName(Class<?> type);

	byte[] buildTypeFromScratch(String newTypeName, Writer writer, IBuildVisitorDelegate buildVisitorDelegate);

	byte[] buildTypeFromParent(String newTypeName, byte[] sourceContent, Writer writer, IBuildVisitorDelegate buildVisitorDelegate);

	Class<?> loadClass(String typeName, byte[] content);

	void verify(byte[] content);
}

File: ambeth/Ambeth.Bytecode/ambeth/bytecode/IBytecodeClassLoader.cs
using System;
using System.Text;

namespace De.Osthus.Ambeth.Bytecode
{
    public interface IBytecodeClassLoader : IBytecodePrinter
    {
	    byte[] ReadTypeAsBinary(Type type);

	    String GetBytecodeTypeName(Type type);

        Type BuildTypeFromScratch(String newTypeName, StringBuilder writer, IBuildVisitorDelegate buildVisitorDelegate);

        Type BuildTypeFromParent(String newTypeName, Type sourceContent, StringBuilder writer, IBuildVisitorDelegate buildVisitorDelegate);

	    Type LoadClass(String typeName, byte[] content);

        void Save();

	    void Verify(byte[] content);
    }
}